### PR TITLE
refactor: Create module with shared utilities

### LINKED ISSUE #1402: Create module with shared utilities
**Author the user story for this feature**

As a Developer, I want to have a shared module so I could better organize and reuse code.

**Describe the solution**
Create `common`/`utils` module

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1402

`:common` modules with shared utils created. Many files has been changed in this PR, however they are mostly import changes

`:common` is available, all tests passed

### CODE REVIEW

File: common/.gitignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Compiled class file
```
Comment (Sloox) [inline common/.gitignore:1]: Can we perhaps use a gitignore generator 
EG : https://www.toptal.com/developers/gitignore

File: common/src/main/kotlin/flank.common/Files.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package flank.common
+
+import com.github.kittinunf.fuel.Fuel
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+val userHome: String by lazy {
+    if (isWindows) System.getenv("HOMEPATH") else System.getProperty("user.home")
```
Comment (Sloox) [inline common/src/main/kotlin/flank.common/Files.kt:10]: Need me to test locally on my machine?

File: common/src/main/kotlin/flank.common/Os.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package flank.common
+
+private val osName = System.getProperty("os.name")?.toLowerCase() ?: ""
+
+val isMacOS: Boolean by lazy {
```
Comment (Sloox) [inline common/src/main/kotlin/flank.common/Os.kt:5]: Do we need a isLinux? or is it the "else" case?

File: common/src/main/kotlin/flank.common/Strings.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.io.StringWriter
+import java.nio.file.Files
+import java.nio.file.Paths
+
+fun String.withNewLineAtTheEnd() = plus(System.lineSeparator())
+
+fun String.normalizeLineEnding(): String {
+    // required for tests to pass on Windows
+    return this.replace("\r\n", "\n")
```
Comment (Sloox) [inline common/src/main/kotlin/flank.common/Strings.kt:11]: 🥇

**Review by Sloox — APPROVED** [review]
Minor comments but looks good 👍

**Comment by pawelpasterz** [comment]
@flank-it

File: common/src/main/kotlin/flank.common/Files.kt (hunk 2 above)
Comment (piotradamczyk5) [reply common/src/main/kotlin/flank.common/Files.kt:10]: I just moved this code, should work properly 🤞

File: common/src/main/kotlin/flank.common/Os.kt (hunk 3 above)
Comment (piotradamczyk5) [reply common/src/main/kotlin/flank.common/Os.kt:5]: basically we just use `isWindows` and `isMacOS`

File: common/src/main/kotlin/flank.common/Strings.kt (hunk 4 above)
Comment (piotradamczyk5) [reply common/src/main/kotlin/flank.common/Strings.kt:11]: 😅

Comment (piotradamczyk5) [reply common/src/main/kotlin/flank.common/Strings.kt:11]: this is just copy-paste code :D

**Review by pawelpasterz — CHANGES_REQUESTED** [review]
Just to prevent automatic merge, until the issue with failing tests is not solved

File: common/.gitignore (hunk 1 above)
Comment (piotradamczyk5) [reply common/.gitignore:1]: hey generated file is too big IMO
https://www.toptal.com/developers/gitignore/api/intellij,kotlin

**Comment by piotradamczyk5** [comment]
@flank-it

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**